omod/src/main/java/org/openmrs/web/controller/PortletController.java

```diff
@@ -195,23 +204,42 @@ public ModelAndView handleRequest(HttpServletRequest request, HttpServletRespons
 					}
 					
 					if (Context.hasPrivilege(PrivilegeConstants.GET_OBS)) {
-						List<Obs> patientObs = Context.getObsService().getObservationsByPerson(p);
-						model.put("patientObs", patientObs);
+						Integer limit = getLimitParameter(request, as, "dashboard.observations.maximumNumberToShow");
+						Integer startIndex = getStartIndexParameter(request);
+						
+						Person person = (Person) p;
+						List<Person> persons = Collections.singletonList(person);
+						
+						// Get paginated observations
+						List<Obs> paginatedObs = Context.getObsService().getObservations(persons, null, null, null, null,
+						    null, Collections.singletonList("obsDatetime desc"), limit, startIndex, null, null, false);
+						
+						// Get all observations for BMI calculation
+						List<Obs> allObs = Context.getObsService().getObservations(persons, null, null, null, null, null,
```

Comment:
If you are loading all the patient's observations from the database into memory, then what optimisation is pagination doing?

Reply:
Enhanced so that:
- Retrieves the latest observation for weight and height for the patient using the same observation retrieval method (getObservations).
- Sorting by obsDatetime desc ensures that only the most recent data is considered.
This has been tested and actually is optimised, loads faster for a patient with > 61k observation_count